src/cryptmd5.ts

```diff
@@ -66,6 +66,17 @@ export class Hash {
     constructor(public value: string) { }
 }
 
+
+export function janus_hash(password: string, salt: string): Hash {
+    if('' === password) {
+        return new Hash('');
+    } else {
+        return crypt_md5(password, salt);
+    }
+}
+
+
+
 /**
```

Comment:
Well, okay, I think you are right. It belongs to node-sds. But calling this a feature is a bit exaggerating 😄 And with call site I did not necessarily meant in a different module.

How about this then?

```ts
public static changeUser(username: string, password: cryptmd5.Hash | ''): Message;
```

This way the type checker can catch if this is an empty string or a cryptmd5.Hash.

```
git --no-pager diff
diff --git a/src/sds.ts b/src/sds.ts
index 5183a51..2ea3ea6 100644
--- a/src/sds.ts
+++ b/src/sds.ts
@@ -246,13 +246,17 @@ export class Message {
      * This message logs in the given user.
      *
      * @param {string} username The user to login. Can be with principal name (e.g., 'duckburg.mickey').
-     * @param {Hash} password The user's password hashed with crypt_md5.
+     * @param {Hash} password The user's password hashed with crypt_md5 or the empty string.
      */
-    public static changeUser(username: string, password: cryptmd5.Hash): Message {
+    public static changeUser(username: string, password: cryptmd5.Hash | ''): Message {
         let msg = new Message();
         msg.add([0, 0, 0, 0, 0, 0, 0, 0, Operation.ChangeUser]);
         msg.addString(ParameterName.User, username);
-        msg.addString(ParameterName.Password, password.value);
+        if (password instanceof cryptmd5.Hash) {
+            msg.addString(ParameterName.Password, password.value);
+        } else {
+            msg.addString(ParameterName.Password, '');
+        }
         return msg;
     }
```